doc/mid-based-signalling.md

```diff
@@ -0,0 +1,252 @@
+        "m.call.transferee": false,
+        "m.call.dtmf": false,
+    },
+    "m.tracks.describe": {
+        "1": { // transceiver mid 1
+            "media_uuid": "cccc-cccc-cccccc-cccc-cccc",
+            "media_group_uuid": "2345-2345-234567-2345-2345",
+            "purpose": "m.usermedia",
+            "kind": "video"
+        },
+        "2": {
+            "media_uuid": "bbbb-bbbb-bbbbbb-bbbb-bbbb",
+            "media_group_uuid": "1234-1234-123456-1234-1234",
+            "purpose": "m.usermedia",
+            "kind": "audio",
+        },
+    },
+}
+```
+
+A focus, however, will not send any tracks by default and therefore does not include an
+`m.tracks.describe` bblock. Instead, it includes an `m.track.advertise` block advertising
+what tracks are available for that `conf_id`.
+
+m.call.answer (focus)
+```
+{
+    "m.negotiate": {
+        "version": 1,
+        "call_id": "35657a5b793ce",
+        "conf_id": "bbe53499f82e3",
+        "party_id": "678910",
+        "description": {
+            "type": "answer",
+            "sdp": "[...]",
+        }
+    },
+    "m.call.capabilities": {
+        "m.call.transferee": false,
+        "m.call.dtmf": false,
+    },
+    "m.tracks.advertise": {
+        "alice:example.org": { // user ID
+            "88888888": { // device ID
+                "2345-2345-234567-2345-2345": [{ // media group uuid
+                    "media_uuid": "aaaa-aaaa-aaaaaa-aaaa-aaaa":
+                    "purpose": "m.usermedia",
+                    "kind": "video",
+                }, {
+                    "media_uuid": "bbbb-bbbb-bbbbbb-bbbb-bbbb":
+                    "purpose": "m.usermedia",
+                    "kind": "audio",
+                },
+            },
+        }
+    },
+}
+```
+
+XXX: How flat vs deep do we want the structure to be here? I've done it quite deep here,
+organised by user ID / device ID / media group UUID, but they could also just be a flat
+list of tracks. It would be more duplication but maybe less effort to read.
+
+The `select_answer` is also tweaked to be more extensible-event like although is essentially
+the same:
+
+m.select\_answer
+```
+"m.select_answer": {
+    "version": "1",
+    "conf_id": "1674732106391mz4ygIc84Q2Z6mJ5",
+    "call_id": "35657a5b793ce",
+    "party_id": "123456",
+    "selected_party_id": "678910",
+}
+```
+
+Once the client receives this, it decides what tracks it wants to receive and then sends
+a subscribe message over the data channel:
+
+m.call.subscribe
+```
+"m.call.subscribe": {
+    "seq": 1,
+    "media_uuids": {
+        "aaaa-aaaa-aaaaaa-aaaa-aaaa": {
+            "width": 1024,
+            "height": 576,
+        },
+        "bbbb-bbbb-bbbbbb-bbbb-bbbb": {},
+    },
+},
+```
+
+This has also been rearrnaged a little to make the media UUIDs the keys and remove the
+unsubscribe section which is unnecessary if we always send the complete set of tracks we
+want to receive (we unsubscribe by just removing the media UUID from the dict).
```

Comment:
I am pretty sure we actually wanted to avoid this to keep the messages short. Perhaps we could use `null` to say we want to unsubscribe. We could use a similar technique for unpublishing

Reply:
Yes, maybe I am barking up the wrong tree with this and it should just be the deltas each time, especially if we have sequence numbers to ensure we don't lose events.